tardis/adapters/sites/moab.py

```diff
@@ -72,13 +72,39 @@ def __init__(self, machine_type: str, site_name: str):
             remote_resource_uuid="JobID", resource_status="State"
         )
 
+        # see job state codes at https://computing.llnl.gov/tutorials/moab/#JobStates
         translator_functions = StaticMapping(
             State=lambda x, translator=StaticMapping(
+                BatchHold=ResourceStatus.Stopped,
+                Canceling=ResourceStatus.Running,
+                CANCELLED=ResourceStatus.Deleted,
+                Completed=ResourceStatus.Deleted,
+                COMPLETED=ResourceStatus.Deleted,
+                COMPLETING=ResourceStatus.Running,
+                Deffered=ResourceStatus.Booting,
+                Depend=ResourceStatus.Error,
+                Dependency=ResourceStatus.Error,
+                FAILED=ResourceStatus.Error,
                 Idle=ResourceStatus.Booting,
+                JobHeldUser=ResourceStatus.Stopped,
+                Migrated=ResourceStatus.Booting,
+                NODE_FAIL=ResourceStatus.Error,
+                NotQueued=ResourceStatus.Error,
+                PENDING=ResourceStatus.Booting,
+                Priority=ResourceStatus.Booting,
+                Removed=ResourceStatus.Deleted,
+                Resources=ResourceStatus.Booting,
                 Running=ResourceStatus.Running,
-                Completed=ResourceStatus.Deleted,
-                Canceling=ResourceStatus.Running,
-                Vacated=ResourceStatus.Stopped,
+                RUNNING=ResourceStatus.Running,
+                Staging=ResourceStatus.Booting,
+                Starting=ResourceStatus.Booting,
+                Suspended=ResourceStatus.Stopped,
+                SUSPENDED=ResourceStatus.Stopped,
+                SystemHold=ResourceStatus.Stopped,
+                TimeLimit=ResourceStatus.Deleted,
+                TIMEOUT=ResourceStatus.Deleted,
+                UserHold=ResourceStatus.Stopped,
+                Vacated=ResourceStatus.Deleted,
             ): translator[x],
             JobID=lambda x: int(x),
```

Comment:
There is no need to wrap types in a lambda. A type is already a callable from its arguments to an instance.

```suggestion
            JobID=int,
```

Reply:
Ah, yes of course. Good catch!